client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -233,6 +233,16 @@ export default function getThankYouPageUrl( {
 		return '/';
 	}
 
+	// Gift purchases need to bypass everything below, especially the updateUrlInCookie
+	// redirection to the ecommerce thank you page for ecommerce plan checkouts.
+	if ( cart?.is_gift_purchase ) {
+		debug( 'gift purchase cart.gift_details', cart?.gift_details );
+		if ( ! cart?.gift_details?.receiver_blog_slug ) {
+			throw new Error( 'Expected receiver_blog_slug in cart.gift_details, slug not found.' );
```

Comment:
Commenting out this field in the backend: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Sqrsnhyg%2Qfubccvat%2Qpneg%2Qwfba%2Qfrevnyvmre.cuc%3Sn%3Qgehr%26e%3Qpsss6qr5%23985-og

Results in the user seeing this error:
![Screenshot 2022-11-09 at 15-09-07 Checkout — WordPress com](https://user-images.githubusercontent.com/811776/200737577-29c4960f-c208-44e6-973d-58a23d698134.png)

And a corresponding console log.

Reply:
Should we add a message to let the gifter know that their checkout was successful?
I guess they would care about that the most even if they don't get the thank you page.

Reply:
As per the discussion here (p1667974250091619/1667968667.485139-slack-CKZHG0QCR), we shouldn't worry to much about it.